.github/workflows/argus-helm-chart-update.yaml

```diff
@@ -0,0 +1,111 @@
+        id: get_app_info
+        uses: chanzuckerberg/argus-artifacts/ci/packages/get-app-info@v0.51.0
+        with:
+          app_name: ${{ inputs.app_name }}
+      - name: Log outputs
+        run: |
+          echo "The environments are: ${{ steps.get_app_info.outputs.app_info }}"
+  update-helm-deps:
+    name: Update Helm Dependencies
+    needs: [fetch-app-info]
+    runs-on: ubuntu-latest
+    if: needs.fetch-app-info.outputs.app_info.envs != '[]'
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      fail-fast: false
+      matrix:
+        env: ${{ fromJson(needs.fetch-app-info.outputs.app_info).envs }}
+    steps:
+      - name: Log info
+        run: |
+          echo "Updating chart in env: ${{ matrix.env.name }} - app root dir: ${{ fromJson(needs.fetch-app-info.outputs.app_info).project_root_path }}"
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup updatecli
+        uses: updatecli/updatecli-action@v2
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.CZI_RELEASE_PLEASE_APP_ID }}
+          private_key: ${{ secrets.CZI_RELEASE_PLEASE_PK }}
+
+      - name: Write updatecli manifest
+        id: write_manifest
+        run: |
+          export UPDATECLI_MANIFEST_FILE=updatecli-${{ matrix.env.name }}.yaml
+          echo "UPDATECLI_MANIFEST_FILE=$UPDATECLI_MANIFEST_FILE" >> $GITHUB_OUTPUT
+          cat <<EOF > $UPDATECLI_MANIFEST_FILE
+          name: Helm chart update
+
+          # Defines how to get "source" information such as Jenkins version
+          sources:
+            stackHelmChart:
+              name: Get the latest Helm chart version
+              kind: helmchart
+              spec:
+                url: https://chanzuckerberg.github.io/argo-helm-charts
+                name: stack
+
+          # Defines "targets" which need to be updated if different than "source" information.
+          targets:
+            helmChartVersions:
+              name: Bump Helm chart versions
+              kind: yaml
+              spec:
+                engine: yamlpath
+                key: $.dependencies[?(@.name == 'stack')].version
+                file: ${{ fromJson(needs.fetch-app-info.outputs.app_info).project_root_path }}/.infra/${{ matrix.env.name }}/Chart.yaml
+          EOF
+
+      - name: Run updatecli
+        id: run_updatecli
+        run: |
+          export UPDATE_LOG_FILE=$(uuidgen)-update.log
+          echo "Running updatecli with manifest ${{ steps.write_manifest.outputs.UPDATECLI_MANIFEST_FILE }}"
+          cat ${{ steps.write_manifest.outputs.UPDATECLI_MANIFEST_FILE }}
+
+          echo "----------------------------------------"
+          updatecli apply --config ${{ steps.write_manifest.outputs.UPDATECLI_MANIFEST_FILE }} >> $UPDATE_LOG_FILE
+          cat $UPDATE_LOG_FILE
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
```

Comment:
it's so clean in https://github.com/chanzuckerberg/github-actions/pull/346/files#diff-f6a84ebe1a7e1622adf71b875d508834ae264c172920fcbfbba03fca4e8195d7R145 I wonder if it would be worth it from a readability standpoint to combine this step w the create PR step and use github.rest.pulls.create

Reply:
I think this step of capturing the output from the shell command to make it available to subsequent steps would still be needed

Reply:
I guess this bash could be rewritten in JS to shell out to updatecli but I think the benefits of how `peter-evans/create-pull-request` updates previous PRs and skips creating a PR when there aren't changes makes it worth keeping in the workflow